### PR TITLE
add step by step guide

### PR DESCRIPTION
**Pull request by baurine**

### CODE REVIEW

**Comment by baurine** [comment]
@dragonly @aylei PTAL, thanks!

**Review by aylei — APPROVED** [review]
lgtm

File: doc/guide.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+(Currently, we need to manually setup them, we try to automate some of them later.)
+
+Required tools for building:
+
+1. TiDB
+
+   - See https://github.com/pingcap/community/blob/master/contributors/README.md for details
+   - [`Go 1.13+`](https://golang.org/doc/install)
+
```
Comment (aylei) [inline doc/guide.md:49]:
```suggestion
   - [Go for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=golang.Go)

```

File: doc/guide.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+   - `make` - Build tool (run common workflows)
+   - `cmake` - Build tool (required for gRPC)
+   - `awk` - Pattern scanning/processing language
+   - `lldb`
+
+1. PD
+
+   - See pd/README.md for details
+   - [`Go 1.13+`](https://golang.org/doc/install)
+
```
Comment (aylei) [inline doc/guide.md:63]:
```suggestion
   - [Go for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=golang.Go)

```

File: doc/guide.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+1. TiKV
+
+   - See tikv/CONTRIBUTING.md for details
+   - [rustup](https://rustup.rs/) - Rust installer and toolchain manager
+   - `make` - Build tool (run common workflows)
+   - `cmake` - Build tool (required for gRPC)
+   - `awk` - Pattern scanning/processing language
+   - `lldb`
+
```
Comment (aylei) [inline doc/guide.md:58]:
```suggestion
   - [LLDB VSCode](https://marketplace.visualstudio.com/items?itemName=lanza.lldb-vscode)

```

**Pull request merged**